pkg/prometheus/agent/operator.go

```diff
@@ -1335,18 +1358,14 @@ func (c *Operator) handleMonitorNamespaceUpdate(oldo, curo interface{}) {
 
 // getMissingPermissions returns the RBAC permissions that the controller would need to be
 // granted to fulfill its mission. An empty map means that everything is ok.
-func (c *Operator) getMissingPermissions(ctx context.Context) (map[string][]string, error) {
-	verbs := map[string][]string{
-		monitoringv1alpha1.PrometheusAgentName:                           {"get", "list", "watch"},
-		fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {"update"},
-	}
+func getMissingPermissions(ctx context.Context, kclient *kubernetes.Clientset, conf operator.Config, verbs map[string][]string) (map[string][]string, error) {
```

Comment:
we cannot move this to k8sutil since it will create cyclic imports, we need to move to another package

Reply:
Which imports are causing the cycle? I'm asking because at first it's not clear

Reply:
we are importing `k8sutil` in operator package and if we move this func to `k8sutil` we need to import operator package over there since we pass operator.Config

Reply:
Since the function only needs `conf.Namespaces.PrometheusAllowList`, we can modify the signature to be 

```suggestion
func getMissingPermissions(ctx context.Context, kclient *kubernetes.Clientset, namespaces []string, verbs map[string][]string) (map[string][]string, error) {
```